lua/telescope/utils.lua

```diff
@@ -3,6 +3,10 @@ local Job = require "plenary.job"
 
 local log = require "telescope.log"
 
+local has_vim_diagnostic = vim.diagnostic ~= nil
+-- TODO why is vim.diagnostic all uppercase..
+local type_diagnostic = vim.lsp.protocol.DiagnosticSeverity
```

Comment:
I guess to match the protocol naming convention

Reply:
Am I missing something? https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#diagnosticSeverity

Reply:
To follow up

> We use our own severity levels to decouple vim.diagnostic from LSP. These are designed similarly to vim.log.levels and currently include: `vim.diagnostic.severity.{ERROR, WARN, INFO, HINT}. In practice, these match the LSP diagnostic severity levels exactly, but we should treat this as an interface and not assume that they are the same. The "translation" between the two severity types is handled transparently in vim.lsp.diagnostic.

As the main post of the original [PR](https://github.com/neovim/neovim/pull/15585) has been edited to explain this. Makes sense, it's just convenient when it lines up with the signs as well :sweat_smile: